mongodbatlasv2/atlas_client.go

```diff
@@ -91,3 +91,16 @@ func UseUserAgent(userAgent string) ClientModifier {
 		return nil
 	}
 }
+
+// ListOptions specifies the optional parameters to List methods that
+// support pagination.
+type ListOptions struct {
```

Comment:
who is the user of this struct?
what do they gain from this?

Reply:
I would say we could would be one of the users for this, for the AtlasCLI migration project as it would result in less work in refactoring the list store methods, i.e., 
```
func (s *Store) Projects(opts *atlas.ListOptions) 
```
would become
```
func (s *Store) Projects(opts *atlasv2.ListOptions) 
```
instead of deciding on a new pattern for this methods across AtlasCLI.

Reply:
I'm fine for us to create a list object but it should not be used as a structure that is marshaled/used as model. 
Using templates for helper structure is something that I will allow, but I would rather us to write go code due to degraded maintenance for template form.

Reply:
>for the AtlasCLI migration project as it would result in less work in refactoring the list store methods

Then why don't we add the code on the CLI?

Reply:
⬆️

Reply:
> Then why don't we add the code on the CLI?
Seemed more like a SDK functionality than a CLI one, since it would also be easier for customers to migrate to the new SDK with list options

> How would you handle cases where list includes more than pagination, things like filters, for example metrics?
If we stick with this, the other parameters could be specified through the current mechanism we have (chaining the parameters on the request). Otherwise, we could have different levels of parameter grouping, that would result in ListOptions for each list method (i.e., `OrganizationListOptions` , `MetricsListOptions` etc.)

Reply:
Food for though, the new SDK uses a fluent interface, `GetThinh().Page().Limit().OtherThing().Execute()` fluent APIs are great and it saves us from a trap we felt (I say "we" even the initial implementation predates anyone in the team) by using a more imperative approach `GetThing(arg1, arg2, argN)`, why a trap, check https://github.com/mongodb/go-client-mongodb-atlas/pull/302, when we get the query params wrong it tends to be a breaking change, now this is why I think the new way is better than the current way and `ListOptions` should be a thing of the past

Now about this PR, it comes with a support cost, once we add it we can't delete, how long do we expect users to use it as a stepping stone to migrate to the new SDK? what do we do if a support case comes in and asks for the equivalent for all related list options since we have a precedence of adding this method here so a bit of a slippery slope

TL;DR I fear adding it here may come with hidden costs, adding to the CLI may be ok, other users may follow a similar approach of internally managing transformations and that's ok, I belive but may need to double check we set a clear goal of not making the new SDK a straight up replacement so we are safe on that side as well

Reply:
Awesome response! Thank you @gssbzn